log/handler.go

```diff
@@ -1,375 +1,290 @@
-	}
-	return values, nil
+func JsonHandler(wr io.Writer) slog.Handler {
+	return &builtinHandler{slog.NewJSONHandler(wr, &slog.HandlerOptions{
+		ReplaceAttr: builtinReplace,
+	})}
 }
 
-// DiscardHandler reports success for all writes but does nothing.
-// It is useful for dynamically disabling logging at runtime via
-// a Logger's SetHandler method.
-func DiscardHandler() Handler {
-	return FuncHandler(func(r *Record) error {
-		return nil
-	})
+// LogfmtHandler returns a handler which prints records in logfmt format, an easy machine-parseable but human-readable
+// format for key/value pairs.
+//
+// For more details see: http://godoc.org/github.com/kr/logfmt
+func LogfmtHandler(wr io.Writer) slog.Handler {
+	return &builtinHandler{slog.NewTextHandler(wr, &slog.HandlerOptions{
+		ReplaceAttr: builtinReplace,
+	})}
 }
 
-// Must provides the following Handler creation functions
-// which instead of returning an error parameter only return a Handler
-// and panic on failure: FileHandler, NetHandler, SyslogHandler, SyslogNetHandler
-var Must muster
+// LogfmtHandlerWithLevel returns the same handler as LogfmtHandler but it only outputs
+// records which are less than or equal to the specified verbosity level.
+func LogfmtHandlerWithLevel(wr io.Writer, level slog.Level) slog.Handler {
+	return &builtinHandler{slog.NewTextHandler(wr, &slog.HandlerOptions{
+		ReplaceAttr: builtinReplace,
+		Level:       &leveler{level},
+	})}
+}
+
+// builtinHandler wraps an instance of the built-in slog json/text handlers
+// a wrapper is needed because with slog.HandlerOptions.ReplaceAttr, there is
+// no way to differentiate between an attribute that is a property of the Record
+// and a user-provided attribute (e.g. providing an attribute who's key is "time"
+// conflicts with the Record.Time property).
+type builtinHandler struct {
+	inner slog.Handler
+}
 
-func must(h Handler, err error) Handler {
-	if err != nil {
-		panic(err)
+func (h *builtinHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
+	return &builtinHandler{
+		h.inner.WithAttrs(attrs),
 	}
-	return h
 }
 
-type muster struct{}
+const diffMarker = "xxxxxx"
 
-func (m muster) FileHandler(path string, fmtr Format) Handler {
-	return must(FileHandler(path, fmtr))
+func (h *builtinHandler) WithGroup(name string) slog.Handler {
+	return &builtinHandler{
+		h.inner.WithGroup(name),
+	}
 }
 
-func (m muster) NetHandler(network, addr string, fmtr Format) Handler {
-	return must(NetHandler(network, addr, fmtr))
+func (h *builtinHandler) Enabled(_ context.Context, level slog.Level) bool {
+	return h.inner.Enabled(context.Background(), level)
 }
 
-// swapHandler wraps another handler that may be swapped out
-// dynamically at runtime in a thread-safe fashion.
-type swapHandler struct {
-	handler atomic.Value
-}
+func (h *builtinHandler) Handle(_ context.Context, r slog.Record) error {
+	var sanitizedAttrs []slog.Attr
+	sanitizedRecord := slog.NewRecord(r.Time, r.Level, r.Message, r.PC)
```

Comment:
I think the `buildinHandler` logic is a bit too convoluted. There's this logic to first detect e..g `time`, conflicting with the 'native' key `time`. And so we change it into `xxxxxxtime` here in `Handle`, and then later on in `builtinReplace` we check every attribute: to see if it is `xxxxxxtime`, if so, swap it back to `time`. 

We jump through a lot of hoops just to make it possible for a developer to use certain keys: `time`, `level`, `msg`, `source`. 

IMO it's not that big of a deal if we were to emit e.g.`_time` or `time_0` or even `xxxxtime` on such a conflict (so don't bother replacing back)

Reply:
I guess what I really mean is: us devs _should not_ use those keys, because it's ambiguous, and won't work well in in `json`, or any format which is mapping-based. So we should just avoid it. But with such fancy / nice workarounds to make it work anyway, we won't even notice if we ever happen to do this. 

Ideally, we'd detect at compilation time that a reserved keyword was used, but, well, can't do that in go :(

Reply:
we could also just remove the builtinHandler entirely and make logfmt/json handlers faster for us as a result.  If we (or users) happen to use conflicting keys, they can deal with the consequences (keys appearing twice in the outputted record).

Reply:
> we could also just remove the builtinHandler entirely and make logfmt/json handlers faster for us as a result. If we (or users) happen to use conflicting keys, they can deal with the consequences (keys appearing twice in the outputted record).

Yes, I think so too. As you can see in my other PR to master, the current json logger already let's the custom fields overwrite the 'native' ones, so that's already broked IMO. I think the simplicity is worth it